### ISSUE TITLE
0.40.1.0 upload errors

### ISSUE THREAD

**Issue by grrrrr** [mozilla/MozStumbler#989]
it is possible I am interpreting the logs but I see constant upload errors on all versions (0.2x, 0.3x, 0.4x).

Most recently I have upgraded to 0.40.1.0 and after a 20 minute mapping session (and 60Mb of data used :( #970 ) I got home and tried to upload the capture.  I pressed the upload and the queue is cleared instantly.

in the logs I see the following

```
<timestamp>: MLS Submit: [HTTP Status:204], [Bytes Sent:241]
<timestamp>: HTTP non-success code: 400, Error, deleting bad report
```

The 2nd line gets repeated 14 times

**Comment by crankycoder** [comment]
That's really strange - what kind of device do you have?

**Comment by grrrrr** [comment]
Galaxy S4 running CM11 M8

**Comment by crankycoder** [comment]
Can you retry this with stock firmware?

**Comment by grrrrr** [comment]
unfortunately I'm not in a position to do that at the moment.

**Comment by crankycoder** [comment]
@grrrrr can you try on v0.40.3.0?

**Comment by grrrrr** [comment]
@crankycoder  sure, it will be a day or two but I'll report back.  I'll be testing on 0.40.4.0.

**Comment by grrrrr** [comment]
so far so good on 0.40.4.0, no error messages in the logs, 8kb sent in less than 1 second

**Comment by AgentSteel** [comment]
Hi, also tried this release 0.40.4.0 and was able to upload some tiny bits (~260 bytes) but still getting several HTTP 400 upload errors, makes me think that only a small fraction of the collected data was uploaded.

In particular, only some wifi hotspots were sent, no cellular antenna (I'm sure at least 1 was detected)

Using stock rom : gingerbread 2.3.6 on Moto Defy+

**Comment by httpless** [comment]
I can confirm that some data is not being uploaded fully from anecdotal evidence.  Following a recording session only some of my data was recorded, despite witnessing mozstumbler recording data in areas which are unmarked in the map.

https://location.services.mozilla.com/map#14/53.3295/-6.2122

In this map, I did Sandymount strand and the main roads of the area marked Irishtown Nature Park.   Prior to the mapping neither areas had been recorded.  This was done as one continuous mapping

On an aside, I have driven the road marked "Sandymount Strand" several times now and this is the first time that the road was added to the map.

I am running similar specs to grrrrr:  Samsung Galaxy S4 with Cyanogen Mod 11 M11 and Mozstumbler 0.40.4.0
